---
title: コンパイラ エラー CS1017
ms.date: 07/20/2015
f1_keywords:
- CS1017
helpviewer_keywords:
- CS1017
ms.assetid: e0902e8a-b042-4711-a8a6-83456a3f88cb
ms.openlocfilehash: 6371673d05ecd427dd410ea639f0f0671f851697
ms.sourcegitcommit: 3d5d33f384eeba41b2dff79d096f47ccc8d8f03d
ms.translationtype: HT
ms.contentlocale: ja-JP
ms.lasthandoff: 05/04/2018
ms.locfileid: "33303658"
---
# <a name="compiler-error-cs1017"></a><span data-ttu-id="b8b71-102">コンパイラ エラー CS1017</span><span class="sxs-lookup"><span data-stu-id="b8b71-102">Compiler Error CS1017</span></span>
<span data-ttu-id="b8b71-103">catch 句を、try ステートメントの一般的な catch 句の後に置くことはできません。</span><span class="sxs-lookup"><span data-stu-id="b8b71-103">Catch clauses cannot follow the general catch clause of a try statement</span></span>  
  
 <span data-ttu-id="b8b71-104">パラメーターを一切受け取らない `catch` ブロックは、一連の `catch` ブロックの最後に配置する必要があります。</span><span class="sxs-lookup"><span data-stu-id="b8b71-104">A `catch` block that does not take any parameters must be the last in a series of `catch` blocks.</span></span> <span data-ttu-id="b8b71-105">例外の詳細については、「 [Exception Handling Statements](../../csharp/language-reference/keywords/exception-handling-statements.md)」を参照してください。</span><span class="sxs-lookup"><span data-stu-id="b8b71-105">For more information on exceptions, see [Exception Handling Statements](../../csharp/language-reference/keywords/exception-handling-statements.md).</span></span>  
  
## <a name="example"></a><span data-ttu-id="b8b71-106">例</span><span class="sxs-lookup"><span data-stu-id="b8b71-106">Example</span></span>  
 <span data-ttu-id="b8b71-107">次の例では CS1017 が生成されます。</span><span class="sxs-lookup"><span data-stu-id="b8b71-107">The following sample generates CS1017:</span></span>  
  
```csharp  
// CS1017.cs  
using System;  
  
namespace x  
{  
    public class b : Exception  
    {  
    }  
  
    public class a  
    {  
        public static void Main()  
        {  
            try  
            {  
            }  
  
            catch   // CS1017, must be last catch  
            {  
            }  
  
            catch(b)  
            {  
                throw;  
            }  
        }  
    }  
}  
```
